docs/README.md

```diff
@@ -1,24 +1,136 @@
-# How To Contribute
+# NGINX Ingress Controller Docs
 
-## Introduction
+This directory contains all of the user documentation for NGINX Ingress Controller, as well as the requirements for linting, building, and publishing the documentation.
 
-The NGINX Ingress Controller makes use of the [Hugo](https://gohugo.io/) static site generator.
+Docs are written in Markdown. We build the docs using [Hugo](https://gohugo.io) and host them on [Netlify](https://www.netlify.com/).
 
-Documentation is stored within the `docs/` subfolder of the main repository, and is where `hugo` commands should be run.
+## Setup
 
-Hugo will watch and reload any changes made, so you can review your work in real time.
+1. To install Hugo locally, refer to the [Hugo installation instructions](https://gohugo.io/getting-started/installing/).
 
-## Set-up
+    > **NOTE**: We are currently running [Hugo v0.115.3](https://github.com/gohugoio/hugo/releases/tag/v0.115.3) in production.
 
-### Requirements
+2. We use markdownlint to check that Markdown files are correct. Use `npm` to install markdownlint-cli:
 
-- [git](https://git-scm.com/downloads)
-- [hugo](https://gohugo.io/installation/)
+    ```shell
+    npm install -g markdownlint-cli
+    ```
 
-### Quick Start**
+## Local Docs Development
 
-```console
-git clone git@github.com:nginxinc/kubernetes-ingress.git
-cd docs/
-hugo server
+To build the docs locally, run the desired `make` command from the docs directory:
+
+```text
+make clean          -   removes the local `public` directory, which is the default output path used by Hugo
+make docs           -   runs a local hugo server so you can view docs in your browser while you work
+make hugo-mod       -   cleans the Hugo module cache and fetches the latest version of the theme module
+make docs-drafts    -   runs the local hugo server and includes all docs marked with `draft: true`
+```
+
+## Linting
+
+- To run the markdownlint check, run the following command from the docs directory:
+
+    ```bash
+    markdownlint -c docs/mdlint_conf.json content
```

Comment:
do we need to add this file `docs/mdlint_conf.json` ?

Reply:
we just recently added `.markdownlint-cli2.yaml` to the root of the project for all other markdown files, but I'm not sure if we want to apply the same rules here for the docs

Reply:
I'll take a look, thanks for catching that!

Reply:
@lucacome I'm taking this out of the README and we will address it in a separate PR. It's a much larger can of worms than I care to bring into the context of this PR.

Reply:
We don't want to use the main markdownlint config file for the docs, but we need to standardize on the rules across all of the repos before going further.